src/main/org/deidentifier/arx/distributed/ARXDistributedAnonymizer.java

```diff
@@ -314,34 +337,44 @@ public ARXDistributedResult anonymize(Data data,
      * @throws RollbackRequiredException 
      * @throws IOException 
      */
-    List<Future<DataHandle>> getAnonymization(List<DataHandle> partitions,
+    List<Future<DataHandle>> getAnonymization(List<ARXPartition> partitions,
                                                       ARXConfiguration config,
                                                       DistributionStrategy distributionStrategy2,
                                                       int[] transformation) throws IOException, RollbackRequiredException {
         List<Future<DataHandle>> futures = new ArrayList<>();
-        for (DataHandle partition : partitions) {
+        for (ARXPartition partition : partitions) {
+            ARXConfiguration localConfig = config.clone();
+            if (!isNullOrEmpty(partition.getSubset())) {
```

Comment:
That you need isNullOrEmpty is likely a consequence of the ugly thing you did above...